src/Refactoring-Core/RBInlineTemporaryRefactoring.class.st

```diff
@@ -38,55 +38,70 @@ RBInlineTemporaryRefactoring class >> model: aRBSmalltalk inline: anInterval fro
 ]
 
 { #category : 'preconditions' }
-RBInlineTemporaryRefactoring >> applicabilityPreconditions [
+RBInlineTemporaryRefactoring >> applicabilityPreconditions [ 
 
+	| validator |
+	
 	^ {
-		  (RBCondition definesSelector: selector in: class).
-		  (RBCondition withBlock: [
-			   self checkSelectedInterval.
-			   true ]) }
+		(ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		
+		(ReBlockCondition new 
+			block: [
+				self checkSelectedInterval.
+				validator := ReIsValidInstanceVariableName new.
+				(validator check: assignmentNode variable name)
+					ifFalse: [ 
+						self refactoringError: (validator violationMessageOn: assignmentNode variable name)].
+				true];
+			violatorErrorString: 'Check selected interval failed').		
```

Comment:
Instead of checking if the condition is satisfied here, maybe it makes more sense just to create an instance of that precondition and put it in the list, same as `ReDefinesSelectorCondition`.
This way we can avoid doing these checks on our own, and let `checkPreconditions` to do them instead of us.